src/util/strencodings.h

```diff
@@ -69,7 +71,25 @@ std::string EncodeBase32(const std::string& str, bool pad = true);
 
 void SplitHostPort(std::string in, int& portOut, std::string& hostOut);
 int64_t atoi64(const std::string& str);
-int atoi(const std::string& str);
+
+template <typename T>
+T ToIntegral(const std::string& str)
+{
+    T result;
+    // Emulate atoi(...) handling of white space and leading +/-.
+    std::string s = TrimString(str);
```

Comment:
Making a copy of the string here, as well as below for the +/- check, is kind of efficient.
Not sure it matters here, it's not like this function is used in any performance critical places at the moment.

Reply:
As a general rule I try to stay away from pointer arithmetic ([rationale here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es42-keep-use-of-pointers-simple-and-straightforward)), that's the reason for the technically extraneous copy which I think is insignificant in this context :)

Personally I find code that avoids pointer arithmetic _much_ easier to reason about from a safety perspective.

That said: I'll let other chime in :)